README.md

```diff
@@ -26,35 +26,120 @@ flutter pub add flame_behaviors
 
 ## Usage ✨
 
-### Behavior
+### Entity
+
+The entity is the building block of a game. It represents a visual game object that can hold 
+multiple `Behavior`s which in turn define how the entity behaves.
+
+```dart
+// Define a custom entity by extending `Entity`.
+class MyEntity extends Entity {
+  MyEntity() : super(behaviors: [MyBehavior()]);
+}
+```
 
-A behavior is a component that defines how an entity behaves. It can be attached to an `Entity` and handle a specific behavior for that entity.
+### Behavior
 
-A behavior can have it's own `Component`s for adding extra functionality related to the behavior. It cannot, however, have its own `Behavior`s.
+A behavior is a component that defines how an entity behaves. It can be attached to an `Entity` 
+and handle a specific behavior for that entity. Behaviors can either be generic for any entity 
+or you can specify the specific type of entity it requires:
```

Comment:
```suggestion
A behavior is a component that defines how an entity behaves. It can be attached to an `Entity` 
and handle a specific behavior for that entity. Behaviors can either be generic for any entity 
or you can specify the specific type of entity that a behavior requires:
```